cypress/e2e/hub/execution-environments.cy.ts

```diff
@@ -1,30 +1,102 @@
 import { hubAPI } from '../../support/formatApiPathForHub';
 import { ExecutionEnvironments } from './constants';
+import { randomString } from '../../../framework/utils/random-string';
+import { RemoteRegistry as IRemoteRegistry } from '../../../frontend/hub/administration/remote-registries/RemoteRegistry';
 
 describe('Execution Environments', () => {
+  const testSignature: string = randomString(5, undefined, { isLowercase: true });
+  function generateRemoteRegistryName(): string {
+    return `test-${testSignature}-remote-registry-${randomString(5, undefined, {
+      isLowercase: true,
+    })}`;
+  }
+
   before(() => {
     cy.hubLogin();
   });
 
-  it('it should render the execution environments page', () => {
+  it('can render the execution environments page', () => {
     cy.navigateTo('hub', ExecutionEnvironments.url);
     cy.verifyPageTitle(ExecutionEnvironments.title);
   });
 
   it('should open a new tab and verify correct docs url', () => {
     cy.navigateTo('hub', ExecutionEnvironments.url);
-
     cy.window().then((win) => {
       cy.stub(win, 'open').as('docsTab');
     });
-
     cy.get('[data-cy="push-container-images"]').click();
-
     cy.get('@docsTab').should(
       'be.calledWith',
       'https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/'
     );
   });
+
+  it('can add and delete a new execution environment', () => {
+    const remoteRegistryName = generateRemoteRegistryName();
+    cy.createRemoteRegistry(remoteRegistryName).then((remoteRegistry: IRemoteRegistry) => {
+      const eeName = `execution_environment_${randomString(3, undefined, { isLowercase: true })}`;
+      const upstreamName = `upstream_name_${randomString(3, undefined, { isLowercase: true })}`;
+
+      cy.navigateTo('hub', ExecutionEnvironments.url);
+      cy.verifyPageTitle(ExecutionEnvironments.title);
+      cy.intercept('GET', hubAPI`/_ui/v1/execution-environments/registries/?limit=50`).as(
+        'registries'
+      );
+      cy.getByDataCy('add-execution-environment').click();
+      cy.wait('@registries')
+        .its('response.body.data.length')
+        .then((count) => {
+          cy.getByDataCy('name').type(eeName);
+          cy.getByDataCy('upstream-name').type(upstreamName);
+          cy.contains('[data-ouia-component-id="menu-select"]', 'Select registry')
+            .click()
+            .then(() => {
+              if (count < 11) {
```

Comment:
It would nice to add a comment explaining this logic here.

Reply:
I took your suggestion, thank you